### DISCUSSION TITLE
Arbitrary RGB_ORDER changes

### DISCUSSION THREAD

**Discussion by Kolodieiev** [Bodmer/TFT_eSPI#2422] [General]
Hi! I am writing my own library for building a graphical interface based on tft_espi .
Here are some defines of my config file
#define ST7789_DRIVER
#define TFT_RGB_ORDER TFT_BGR
#define TFT_WIDTH 240
#define TFT_HEIGHT 320
#define LOAD_FONT2

The problem is this. When I write views, I need to test them some times. Uploading the code to my esp32. The colors on the display are inverted, although everything else works as expected. I could chalk it up to cosmic radiation while setting the RGB_order flag on the display, but reloading the code doesn't change anything. I go to the driver settings file, change the TFT_RGB_ORDER to the opposite one. The colors on the display begin to appear correctly. I continue to write my library and at some point the colors start to invert again! I'm just based on the possibilities that tft_espi provides, I write my own views, how can this affect the rgb sorting order? I realize that I am providing little information, but does anyone have any idea why this is happening? And yes, there is nothing wrong with that. When my library will complete, I'll just set the color sort order I want. Right now it's just a little hindrance while testing my code. But I would like to understand why this is happening.

P.S. I apologize for my English if something is wrong!

**Comment by Bodmer** [comment]
Since this is your own code that is causing problems it is not possible to identify the root cause.  Some guesses are: spurioius data being sent, incorrect DC signal timing, SPI too fast, software bug,
